File: LeetCode 热题 HOT 100/二叉树/144. 二叉树的前序遍历（递归）.py
# 给你二叉树的根节点 root ，返回它节点值的 前序 遍历。

# 示例 1：
# 输入：root = [1, null, 2, 3]
# 输出：[1, 2, 3]

# 示例 2：
# 输入：root = []
# 输出：[]

# 示例 3：
# 输入：root = [1]
# 输出：[1]

# 示例 4：
# 输入：root = [1, 2]
# 输出：[1, 2]

# 示例 5：
# 输入：root = [1, null, 2]
# 输出：[1, 2]

# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

def preorderTraversal(root):
    res = []

    def getVal(root):
        if root:
            res.append(root.val)
            getVal(root.left)
            getVal(root.right)
    getVal(root)
    return res

File: LeetCode 热题 HOT 100/二叉树/144. 二叉树的前序遍历（递归）.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var preorderTraversal = function (root) {
    res = []
    function getVal(root) {
        if (root !== null) {
            res.push(root.val)
            getVal(root.left)
            getVal(root.right)
        }
    }

    getVal(root)
    return res
};